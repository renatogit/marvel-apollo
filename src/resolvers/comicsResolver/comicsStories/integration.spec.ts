const DataSourcesComicsStoriesAPI = require('@/dataSources/comics');
const ComicsStoriesResolver = require('.');
const ServerGetComicsStories = require('@/configs/test');
const QUERY_GET_COMICS_STORIES = require('@query/comics/queryComicsStories');
const MOCK_GET_COMICS_STORIES = require('@mock/comics/mockComicsStories');

describe('resolvers/comicsStoriesResolver/integration', () => {
	const comicsId = {comicsId: 'fake-code'};

	it('fetches comicsStories and generate snapshots', async () => {
		const {server} = await ServerGetComicsStories(
			DataSourcesComicsStoriesAPI,
			ComicsStoriesResolver,
			MOCK_GET_COMICS_STORIES,
			comicsId
		);

		const {body} = await server.executeOperation({
			query: QUERY_GET_COMICS_STORIES,
			variables: comicsId,
		});

		expect(body).toMatchSnapshot();
	});
});
