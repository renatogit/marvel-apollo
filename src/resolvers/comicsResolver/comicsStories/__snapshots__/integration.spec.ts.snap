// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/comicsStoriesResolver/integration fetches comicsStories and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "comicsStories": {
        "attributionHTML": "mock data",
        "attributionText": "mock data",
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 2,
          "limit": 20,
          "offset": 0,
          "results": [
            {
              "characters": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "comics": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "creators": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "description": "mock data",
              "events": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "id": 21328,
              "modified": "mock data",
              "originalissue": {
                "name": "mock data",
                "resourceURI": "mock data",
              },
              "resourceURI": "mock data",
              "series": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "title": "mock data",
              "type": "mock data",
            },
            {
              "characters": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "comics": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "creators": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "description": "mock data",
              "events": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "id": 21329,
              "modified": "mock data",
              "originalissue": {
                "name": "mock data",
                "resourceURI": "mock data",
              },
              "resourceURI": "mock data",
              "series": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "title": "mock data",
              "type": "mock data",
            },
          ],
          "total": 2,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
